presto-hive/src/main/java/io/prestosql/plugin/hive/HiveLocationService.java

```diff
@@ -55,7 +55,7 @@ public HiveLocationService(HdfsEnvironment hdfsEnvironment)
     public LocationHandle forNewTable(SemiTransactionalHiveMetastore metastore, ConnectorSession session, String schemaName, String tableName, Optional<Path> externalLocation)
     {
         HdfsContext context = new HdfsContext(session, schemaName, tableName);
-        Path targetPath = externalLocation.orElse(getTableDefaultLocation(context, metastore, hdfsEnvironment, schemaName, tableName));
```

Comment:
sigh. i wish `orElse` had longer name, so that it would be simply more convenient to call `orElseGet`.

I guess you can find many places like that in the codebase.

Reply:
I'll add tests to exercise this check later - meanwhile you can ignore this PR. So true about Optional, I myself was surprised seeing this behaviour.